common/account_utils/src/validator_definitions.rs

```diff
@@ -113,18 +128,31 @@ impl ValidatorDefinition {
 
 /// A list of `ValidatorDefinition` that serves as a serde-able configuration file which defines a
 /// list of validators to be initialized by this validator client.
-#[derive(Default, Serialize, Deserialize)]
-pub struct ValidatorDefinitions(Vec<ValidatorDefinition>);
+pub struct ValidatorDefinitions {
+    definitions: Vec<ValidatorDefinition>,
+    _lockfile: Lockfile,
+}
 
 impl ValidatorDefinitions {
+    pub fn create<P: AsRef<Path>>(validators_dir: P) -> Result<Self, Error> {
+        Ok(Self {
+            definitions: vec![],
+            _lockfile: Lockfile::new({
+                let mut path: PathBuf = validators_dir.as_ref().to_path_buf();
+                path.set_extension(LOCK_FILE);
+                path
```

Comment:
I see you've switched to `set_extension` instead of `join`, because `join`ing adds a subdirectory like `./validator_definitions.yml/.lock`, which I hadn't noticed initially.

However, this is calling `set_extension` on the _directory_, which is a no-op, and then trying to lock the directory silently succeeds... Instead I think we should change `LOCK_FILE` to be `.validator_definitions.yml.lock` (consistent with our other naming conventions), and use `join` on the `validators_dir` like:

```rust
_lockfile: Lockfile::new(validators_dir.as_ref().join(LOCK_FILE))?
```

Reply:
@michaelsproul Alternative is `Path::with_extension`, which returns a `PathBuf`, although I agree it's probably just simpler to change the definition of `LOCK_FILE` as it would avoid this bug altogether.

Reply:
Yeah I think `with_extension` is also a bit unweildy because it swaps the file extension, so calling it on the config path would yield `validator_definitions.lock`

Reply:
Consult 75af6f74.